bottlecap/src/logs/agent.rs

```diff
@@ -0,0 +1,76 @@
+use std::sync::mpsc::{self, Sender};
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+use tracing::{debug, error};
+
+use crate::config;
+use crate::logs::aggregator::Aggregator;
+use crate::logs::datadog;
+use crate::logs::processor::Processor;
+use crate::telemetry::events::TelemetryEvent;
+
+pub struct LogsAgent {
+    dd_api: datadog::DdApi,
+    aggregator: Arc<Mutex<Aggregator>>,
+    tx: Sender<TelemetryEvent>,
+    join_handle: std::thread::JoinHandle<()>,
+}
+
+impl LogsAgent {
+    pub fn run(function_arn: &str, datadog_config: Arc<config::Config>) -> LogsAgent {
+        let function_arn = function_arn.to_string();
+        let aggregator: Arc<Mutex<Aggregator>> = Arc::new(Mutex::new(Aggregator::default()));
+        let processor: Processor = Processor::new(function_arn, Arc::clone(&datadog_config));
+
+        let cloned_aggregator = Arc::clone(&aggregator);
+
+        let (tx, rx) = mpsc::channel::<TelemetryEvent>();
+        let join_handle = thread::spawn(move || loop {
+            let received = rx.recv();
+            if let Ok(event) = received {
```

Comment:
Stylistically you can write this as:
```
let Ok(event) = received else {
   break;
}
```
This let's us avoid the creeping indent problem. You also never break from this loop, even when we start getting errors from the channel, (like when it's closed).

Reply:
Resolved!